connman/vpn/vpn-provider.c

```diff
@@ -3463,6 +3469,62 @@ int vpn_provider_append_route(struct vpn_provider *provider,
 	return 0;
 }
 
+int vpn_provider_append_route_complete(struct vpn_provider *provider,
+				unsigned long idx, int family,
+				const char *network, const char *netmask,
+				const char *gateway)
+{
+	struct vpn_route *route;
+
+	DBG("provider %p idx %lu family %d network %s netmask %s gateway %s",
+						provider, idx, family, network,
+						netmask, gateway);
+
+	route = provider_lookup_route(provider, idx, family);
+
+	route->network = g_strdup(network);
+	route->netmask = g_strdup(netmask);
+	route->gateway = g_strdup(gateway);
```

Comment:
Is there any chance that already set values would be overwritten here?

Reply:
This follows the same style that is used with other plugins, where they append the routes gradually. So, if the plugin does add the routes with the `idx` that is not going to be unique, which should never be done, it is doing this wrong and there will be leaks. The purpose here would be just add a new route, not sure if this needs to do it different from the other, to reply `-EALREADY` when a route with matching `idx`  is found.

Reply:
Well, with 43ad8b65e480773bf208f535e85099ca4b71bac1 this is now a lot cleaner. Resolving this as.